### ISSUE TITLE
bug(android/engine): Duplicate Menu Entry in Settings > Installed Languages > "+"

### ISSUE THREAD

**Issue by dyacob** [keymanapp/keyman#3806]
**Describe the bug**
Launching the Keyman app this morning a message briefly appeared indicating that a catalog had updated. Navigating to install a new language the menu listing showed 2 "Sebat Bet Gurage" entries.  The first offered two keyboards, clicking the duplicate entry an error message appears "One or more resource items failed to update".

**To Reproduce**
1. Go to Settings > Installed Languages > +
2. Scroll down to "Sebat Bet Gurage"
4. See error, the 2nd entry is erroneous.

**Expected behavior**
Only a single entry should appear.

**Screenshots**

![Double-Sebat-Bet](https://user-images.githubusercontent.com/8098333/97710877-b6d6a600-1a92-11eb-8070-5718c570dee2.jpg)

---
  
**iPhone/iPad/Android:**
 - Device: Samsung S9
 - OS: Android 10
 - Keyman version: 13.0.6219
 - Target application: Keyman

---
 
**Additional context**
I expect this is related to the recent deprecation of the gff_sgw_7 keyboard.  There may have also been an "sgw" vs "sgw-Ethi" difference between these keyboards.

If at all possible, it would be easier for users to find the keyboard if the encompassing menu was simply "Gurage".  If "Sebat Bet" must remain, then presenting it as "Gurage, Sebat Bet" would be better.

**Comment by darcywong00** [comment]
I'm documenting my steps to reproduce this issue on a fresh install of Keyman for Android 13.0.6220 stable

Initially, the **Add New Language** menu only lists **Sebat Bet Gurage (Ethiopic)**.
That language has two **Gurage** keyboards available.
When I install the first one,  I see **Sebat Bet Gurage** gets added to the language menu. 
Clicking on **Sebat Bet Gurage** gives me the "One or more resource failed to update!" error.

> If at all possible, it would be easier for users to find the keyboard if the encompassing menu was simply "Gurage".

We'll be improving the entire keyboard-search experience in 14.0. Here's what you'd see in Keyman for Android 14.0 alpha
![Screenshot_1604281166](https://user-images.githubusercontent.com/7358010/97822096-470d2900-1ce7-11eb-9c33-914c9e3c43c7.png)

I'll label this issue as an Android bug while I investigate. The 13.0 search is powered by [api.keyman.com](https://github.com/keymanapp/api.keyman.com) so this issue might get moved to that repo.

**Comment by darcywong00** [comment]
Ok, here's what I think is going on in 13.0:

Your current **gff_gurage** keyboard is downloaded via this api.keyman.com [query](https://api.keyman.com/cloud/4.0/languages/sgw-ethi/gff_gurage?version=13.0.0&device=androidphone&languageidtype=bcp47
) where the language ID is `sgw-ethi` for **Sebat Bet Gurage (ethiopic)**.

When the keyboard is downloaded, we're also checking and installing the associated lexical-model of `sgw-ethi` with this api.keyman.com [query](https://api.keyman.com/model?q=bcp47:sgw-ethi). So the Gurage dictionary package (`gff.sgw.gurage.model.kmp`) gets installed. The query info comes from [gff.sgw.gurage.model_info](https://github.com/keymanapp/lexical-models/blob/master/release/gff/gff.sgw.gurage/gff.sgw.gurage.model_info).

**But**, the dictionary package gets installed on the device for the language **Sebat Bet Gurage** with a language ID `sgw` instead of `sgw-ethi`) as defined in [gff.sgw.gurage.model.kps](https://github.com/keymanapp/lexical-models/blob/master/release/gff/gff.sgw.gurage/source/gff.sgw.gurage.model.kps). 

This extra **Gurage** language (`sgw`) gets merged in the language list, but `sgw` no longer has available keyboards to install.
That sgw [query](https://api.keyman.com/cloud/4.0/languages/sgw?version=13.0.0&device=androidphone&languageidtype=bcp47) returns the deprecated gff_sgw_ keyboard.

Even though `sgw` and `sgw-Ethi` are equivalent tags, the mobile apps are currently limited in matching the entire language tag.

----

TL;DR - The language ID for the keyboard gff_gruage and the lexical-model gff.sgw.gurage need to be consistent (`sgw` or `sgw-ethi`). From your [comment](https://github.com/keymanapp/lexical-models/pull/76#discussion_r421202519) on the lexical-model PR
> I reviewed other GFF keyboards and to be consistent with them the "-Ethi" should be removed from the gff_gurage keyboard. The gff_gurage keyboard does not have a tailored Touch Layout, I'll add one this weekend and make the identifier change.

It sounds like you'll want to update the gff_gurage keyboard to use `sgw` and hopefully that doesn't impact the desktop keyboard installation (in case Windows requires `sgw-Ethi`)

**Comment by darcywong00** [comment]
The Language List is going away in 14.0.
But a possible fix for 13.0 is to remove a language from the list if there's only deprecated keyboards for it?

**Comment by darcywong00** [comment]
@dyacob 
In our team discussion, we decided we won't be fixing this issue for 13.0 primarily because the Language List is going away in 14.0 and the keyboard search should be straight forward with the search bar.

As for the gff_gurage lexical-model, I recommend including both `sgw` and `sgw-Ethi` language codes in the Package Editor (.kps file) and the .model_info file.
